### PR TITLE
ToscaMetadataProvider handles all ToscaTypeProviders

### PR DESCRIPTION
**Pull request by sjcorbett**

### CODE REVIEW

**Comment by sjcorbett** [comment]
@robertgmoss Think this simplifies things a bit. Could you review?

**Comment by robertgmoss** [comment]
Through the user interface, higher priority providers are added to the end of the list.  Either reverse the list or add to front when constructing providers.

**Comment by sjcorbett** [comment]
Better?  https://github.com/cloudsoft/brooklyn-tosca/pull/33/files#diff-26823eef8b44cf151864f9ff6a3992e3R117

File: a4c-brooklyn-plugin/src/main/java/alien4cloud/brooklyn/BrooklynOrchestratorFactory.java [hunk 1]
```diff
@@ -43,12 +46,14 @@ public void destroy(BrooklynOrchestrator instance) {
 
     @Override
     public Configuration getDefaultConfiguration() {
+        List<String> metadataProviders = MutableList.of(
+                BrooklynToscaTypeProvider.class.getName(),
+                DefaultToscaTypeProvider.class.getName());
```
Comment (robertgmoss) [inline a4c-brooklyn-plugin/src/main/java/alien4cloud/brooklyn/BrooklynOrchestratorFactory.java]: These are in the wrong order - DefaultToscaTypeProvider should be lowest priority i.e first in this list

Comment (sjcorbett) [inline a4c-brooklyn-plugin/src/main/java/alien4cloud/brooklyn/BrooklynOrchestratorFactory.java]: Thanks

**Pull request merged**